music_assistant/providers/ibroadcast/__init__.py

```diff
@@ -0,0 +1,469 @@
+        for track in (await self._client.get_tracks()).values():
+            try:
+                yield await self._parse_track(track)
+            except IndexError:
+                continue
+            except (KeyError, TypeError, InvalidDataError) as error:
+                self.logger.debug("Parse track failed: %s", track, exc_info=error)
+                continue
+
+    def _get_artist_item_mapping(self, artist_id, artist_obj: dict) -> ItemMapping:
+        if (not artist_id and artist_obj["name"] == "Various Artists") or artist_id == 0:
+            artist_id = VARIOUS_ARTISTS_MBID
+        return self._get_item_mapping(MediaType.ARTIST, artist_id, artist_obj.get("name"))
+
+    def _get_item_mapping(self, media_type: MediaType, key: str, name: str) -> ItemMapping:
+        return ItemMapping(
+            media_type=media_type,
+            item_id=key,
+            provider=self.instance_id,
+            name=name,
+        )
+
+    async def get_library_playlists(self) -> AsyncGenerator[Playlist, None]:
+        """Retrieve playlists from iBroadcast."""
+        for playlist in (await self._client.get_playlists()).values():
+            # Skip the auto generated playlist
+            if playlist["type"] != "recently-played" and playlist["type"] != "thumbsup":
+                yield await self._parse_playlist(playlist)
+
+    async def get_playlist(self, prov_playlist_id: str) -> Playlist:
+        """Get full playlist details by id."""
+        playlist_obj = await self._client.get_playlist(int(prov_playlist_id))
+        try:
+            playlist = await self._parse_playlist(playlist_obj)
+        except (KeyError, TypeError, InvalidDataError, IndexError) as error:
+            self.logger.debug("Parse playlist failed: %s", playlist_obj, exc_info=error)
+        return playlist
+
+    async def get_playlist_tracks(self, prov_playlist_id: str, page: int = 0) -> list[Track]:
+        """Get playlist tracks."""
+        tracks: list[Track] = []
+        if page > 0:
+            return tracks
+
+        playlist_obj = await self._client.get_playlist(int(prov_playlist_id))
+
+        if "tracks" not in playlist_obj:
+            return tracks
+
+        return await self._get_tracks(playlist_obj["tracks"], True)
+
+    async def get_stream_details(self, item_id: str) -> StreamDetails:
+        """Return the content details for the given track when it will be streamed."""
+        # How to buildup a stream url:
+        # [streaming_server]/[url]?Expires=[now]&Signature=[user token]&file_id=[file ID]
+        # &user_id=[user ID]&platform=[your app name]&version=[your app version]
+        # See https://devguide.ibroadcast.com/?p=streaming-server
+
+        url = await self._client.get_full_stream_url(int(item_id), "music-assistant")
+
+        return StreamDetails(
+            provider=self.instance_id,
+            item_id=item_id,
+            audio_format=AudioFormat(
+                content_type=ContentType.MPEG,
```

Comment:
This is always MPEG ? If you are not sure you can use unknown here which instructs ffmpeg to work it out

Reply:
So far I noticed, but lets switch, just in case